server/routes/authentication.ts

```diff
@@ -1,89 +1,65 @@
+            where: { token: query.token.toString() },
+            attributes: ['userId', 'token', 'expiresIn'],
+        });
+        const [current, expiration] = [Date.now(), new Date(expiresIn).getTime()];
+        const determineMessageStatus: () => [string, number] = () => {
+            switch (true) {
+                case !token:
+                    return ['Token not found', 401];
+                case expiration <= current:
+                    return ['Token expired', 401];
+                default:
+                    return ['The account has been verified. Please log in.', 201];
             }
-            return res.status(status).send({ message });
-        } catch (err) {
-            next(err);
-            return res.status(503).send({ message: 'Server error, try again later' });
+        };
+        const [message, status] = determineMessageStatus();
+
+        if (status === 201) {
+            const user = await db.User.findOne({ where: { id: userId }, attributes: ['isVerified'] });
+            await user.update({ id: userId, isVerified: true });
         }
-    },
-);
+        return res.status(status).send({ message });
+    } catch (err) {
+        next(err);
+        return res.status(503).send({ message: 'Server error, try again later' });
+    }
+});
 
 router.post('/resendConfirm', async (req: Request, res: Response, next: NextFunction) => {
     try {
         const { headers } = req;
         const { email } = req.body;
-        const { id: userId } = await db.User.findOne({
-            where: { email },
-            attributes: ['id'],
-        });
-        const userToken = await db.Token.findOne({
-            where: { userId },
-            attributes: ['id', 'token', 'expiresIn'],
-        });
+        const { id: userId } = await db.User.findOne({ where: { email }, attributes: ['id'] });
+        const userToken = await db.Token.findOne({ where: { userId }, attributes: ['id', 'token', 'expiresIn'] });
         const newToken = crypto.randomBytes(16).toString('hex');
         // update token entry with new token and extended expire time
-        userToken.update({
-            id: userToken.id,
-            token: newToken,
-            expiresIn: Date.now() + 30 * 60 * 1000,
-        });
-        const [message, status] = helper.sendVerEmail(
-            email,
-            headers,
-            newToken,
-            'confirmation',
-        );
-        return res.status(status).send(message);
+        userToken.update({ id: userToken.id, token: newToken, expiresIn: Date.now() + 30 * 60 * 1000 });
+        const [message, status] = sendVerEmail(email, headers, newToken, 'confirmation');
+        return res.status(status).send({ message });
     } catch (err) {
-        next(err);
-        return res.status(503).send({ message: 'Server error, try again later' });
+        res.status(503).send({ message: 'Server error, try again later' });
+        return next(err);
```

Comment:
i think you mean this:
```suggestion
        next(err);
        return res.status(503).send({ message: 'Server error, try again later' });
```
i don't remember how `next()` exactly works in express, but it's different elsewhere in this PR, for example `server/routes/notifications.ts`: https://github.com/House-on-a-Rock/tjc-scheduling-server/pull/18/files#diff-74513d3474f0309bc6da7f843cd531efR34-R38

Reply:
ahh good look thanks